### PR TITLE
drivers: more `struct spi_dt_spec` conversions

### PR DESCRIPTION
**Pull request by JordanYates**
Convert some more drivers to newer devicetree API's.

### CODE REVIEW

**Review by gmarull — APPROVED** [review]
LGTM, two minor comments. Thanks for these cleanups!

**Review by mniestroj — APPROVED** [review]

File: drivers/eeprom/eeprom_at2x.c [hunk 1]
```diff
@@ -35,13 +35,12 @@ LOG_MODULE_REGISTER(eeprom_at2x);
...
 struct eeprom_at2x_config {
-	const struct device *bus_dev;
 	union {
 #ifdef CONFIG_EEPROM_AT24
-		uint16_t i2c_addr;
+		struct i2c_dt_spec i2c;
 #endif /* CONFIG_EEPROM_AT24 */
-#ifdef CONFIG_EEPROM_AT25
-		struct spi_config spi_cfg;
+#if CONFIG_EEPROM_AT25
```
Comment (henrikbrixandersen) [inline drivers/eeprom/eeprom_at2x.c:42]: Why change `#ifdef` to `#if` here?

File: drivers/eeprom/eeprom_at2x.c [hunk 2]
```diff
@@ -260,7 +267,7 @@ static int eeprom_at24_read(const struct device *dev, off_t offset, void *buf,
 	timeout = k_uptime_get() + config->timeout;
 	while (1) {
 		int64_t now = k_uptime_get();
-		err = i2c_write_read(config->bus_dev, bus_addr,
+		err = i2c_write_read(config->bus.i2c.bus, bus_addr,
```
Comment (henrikbrixandersen) [inline drivers/eeprom/eeprom_at2x.c:270]: Could we use `i2c_write_read_dt()` here?

File: drivers/eeprom/eeprom_at2x.c [hunk 3]
```diff
@@ -308,7 +315,7 @@ static int eeprom_at24_write(const struct device *dev, off_t offset,
 	timeout = k_uptime_get() + config->timeout;
 	while (1) {
 		int64_t now = k_uptime_get();
-		err = i2c_write(config->bus_dev, block, sizeof(block),
+		err = i2c_write(config->bus.i2c.bus, block, sizeof(block),
```
Comment (henrikbrixandersen) [inline drivers/eeprom/eeprom_at2x.c:318]: Could we use `i2c_write_dt()` here?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
A few minors on the EEPROM change.

File: drivers/eeprom/eeprom_at2x.c (hunk 2 above)
Comment (JordanYates) [reply drivers/eeprom/eeprom_at2x.c:270]: No, because `bus_addr` is not the raw reg address of the node, it is derived in `eeprom_at24_translate_offset`

**Review by bbilas — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]
EEPROM part reviewed, thanks.

**Pull request merged**